### ISSUE TITLE
Testing a branch in vagrant

### ISSUE THREAD

**Issue by 153957** [HiSPARC/publicdb#95]
Im trying to test my django-1.7 branch.
However, when I run provision the [master branch is checked out](https://github.com/HiSPARC/publicdb/blob/master/provisioning/publicdb.yml#L19-L21) in the VM.
The git module supports a [`version`](http://docs.ansible.com/git_module.html#options) argument that can be set to a tag, hash, or branch name.
I may not be familiar enough with ansible (and a bit of Googling didn't find an answer), but can we make it so that the VM checks out the currently active branch? (`git rev-parse --abbrev-ref HEAD`)
Of course this does mean that the branch you want to test has to be published on GitHub (and up to date). Or, instead of the active branch the current commit (`git rev-parse HEAD`), which would also need to be synced to GitHub..

Any ideas?

@davidfokkema

**Comment by 153957** [comment]
Perhaps something like this to use the last commit from the local repo:

```
cdpub
git remote add local /vagrant/.git
git fetch local
git checkout $(cd /vagrant; git rev-parse --verify HEAD)
```

This will require commits to be made, but does not require them to be synced to GitHub (so can be undone/amended/changed).

**Comment by 153957** [comment]
Related: #71

**Comment by 153957** [comment]
Fixed in ab5a815f5630042aabbb9be0c80c4a732e7bb385 and 36ee953bb4f18880db045677364bf7a5915c9096.

These commits add the `adddev` and `updev` aliases.
Run `adddev` once to add the local repo as a git remote.
Then work in your local copy, commit things you want to test.
Run `updev` in the VM to fetch and checkout those changes (also collectstatic and reload uWSGI).
If you are unhappy with the commits you can undo the commit,
make changes, recommit, and run `updev` again.
As long as you do not push the commit (to GitHub) you can simply undo it and make new changes if you desire..

Perhaps not the perfect solution, but better than it was.

**Comment by 153957** [comment]
Further fixed in 6acbbdc, by cloning the local publicdb repo instead of the GitHub one.
The local HEAD is checked out in the VM.
